hera_cal/abscal.py

```diff
@@ -981,510 +2636,319 @@ def TT_Phi_gain_arr(self):
+        all_model_times, all_model_lsts = get_all_times_and_lsts(hdm, solar_horizon=model_solar_horizon, unwrap=True)
+        d2m_time_map = get_d2m_time_map(all_data_times, all_data_lsts, all_model_times, all_model_lsts)
+        
+        # group matched time indices for partial I/O
+        matched_tinds = [tind for tind, time in enumerate(hd.times) if time in d2m_time_map and d2m_time_map[time] is not None]
+        if len(matched_tinds) > 0:
+            tind_groups = np.array([matched_tinds])  # just load a single group
+            if nInt_to_load > 0:  # split up the integrations to load nInt_to_load at a time
+                tind_groups = np.split(matched_tinds, np.arange(nInt_to_load, len(matched_tinds), nInt_to_load))
+
+            # loop over polarizations
+            for pol in pol_load_list:
+                echo('\n\nNow calibrating ' + pol + '-polarization...', verbose=verbose)
+                # loop over groups of time indices
+                for tinds in tind_groups:
+                    echo('\n    Now calibrating times ' + str(hd.times[tinds[0]])
+                         + ' through ' + str(hd.times[tinds[-1]]) + '...', verbose=verbose)
+                    
+                    # load data
+                    data, flags, nsamples = hd.read(times=hd.times[tinds], polarizations=[pol])
+                    if not np.all(flags.values()):
+                        # load model and rephase
+                        model_times_to_load = [d2m_time_map[time] for time in hd.times[tinds]]
+                        model, model_flags, _ = io.partial_time_io(hdm, model_times_to_load, polarizations=[pol])
+                        model_bls = {bl: model.antpos[bl[0]] - model.antpos[bl[1]] for bl in model.keys()}
+                        utils.lst_rephase(model, model_bls, model.freqs, data.lsts - model.lsts,
+                                          lat=hdm.telescope_location_lat_lon_alt_degrees[0], inplace=True)
+                        
+                        # update data flags w/ model flags
+                        for k in flags.keys():
+                            if k in model_flags:
+                                flags[k] += model_flags[k]
+
+                        # apply calibration
+                        data_ants = set([ant for bl in data.keys() for ant in split_bl(bl)])
+                        rc_gains_subset = {k: rc_gains[k][tinds, :] for k in data_ants}
+                        rc_flags_subset = {k: rc_flags[k][tinds, :] for k in data_ants}
+                        calibrate_in_place(data, rc_gains_subset, data_flags=flags, 
+                                           cal_flags=rc_flags_subset, gain_convention=hc.gain_convention)
+                        
+                        # run absolute calibration, copying data because it gets modified internally
+                        delta_gains, AC = post_redcal_abscal(model, copy.deepcopy(data), flags, rc_flags_subset, edge_cut=edge_cut, 
```

Comment:
lets think if this deepcopy is really needed. I don't think it is. Just use `AC.data` and get rid of second `calibrate_in_place`

Reply:
The issue here is that I need calibrated autocorrelations to normalize chi^2... which are often deleted from AC.data. However, I think I can solve the issue in a memory-efficient way by deepcopying the autos before creating the Abscal object, rather than deepcopying the whole dataset. Thanks for making me think it through.